worlds/doom_1993/Rules.py

```diff
@@ -0,0 +1,513 @@
+
+    progression_items = [
+        "Hangar",
+        "Nuclear Plant",
+        "Nuclear Plant - Red keycard",
+        "Toxin Refinery",
+        "Toxin Refinery - Yellow keycard",
+        "Toxin Refinery - Blue keycard",
+        "Command Control",
+        "Command Control - Yellow keycard",
+        "Command Control - Blue keycard",
+        "Phobos Lab",
+        "Phobos Lab - Blue keycard",
+        "Phobos Lab - Yellow keycard",
+        "Central Processing",
+        "Central Processing - Blue keycard",
+        "Central Processing - Red keycard",
+        "Central Processing - Yellow keycard",
+        "Computer Station",
+        "Computer Station - Yellow keycard",
+        "Computer Station - Blue keycard",
+        "Computer Station - Red keycard",
+        "Phobos Anomaly",
+        "Military Base",
+        "Military Base - Yellow keycard",
+        "Military Base - Red keycard",
+        "Military Base - Blue keycard",
+        "Deimos Anomaly",
+        "Deimos Anomaly - Blue keycard",
+        "Deimos Anomaly - Red keycard",
+        "Containment Area",
+        "Containment Area - Blue keycard",
+        "Containment Area - Yellow keycard",
+        "Containment Area - Red keycard",
+        "Refinery",
+        "Refinery - Blue keycard",
+        "Deimos Lab",
+        "Deimos Lab - Blue keycard",
+        "Deimos Lab - Yellow keycard",
+        "Command Center",
+        "Halls of the Damned",
+        "Halls of the Damned - Blue skull key",
+        "Halls of the Damned - Yellow skull key",
+        "Halls of the Damned - Red skull key",
+        "Spawning Vats",
+        "Spawning Vats - Red keycard",
+        "Spawning Vats - Yellow keycard",
+        "Spawning Vats - Blue keycard",
+        "Tower of Babel",
+        "Fortress of Mystery",
+        "Fortress of Mystery - Blue skull key",
+        "Fortress of Mystery - Red skull key",
+        "Fortress of Mystery - Yellow skull key",
+        "Hell Keep",
+        "Slough of Despair",
+        "Slough of Despair - Blue skull key",
+        "Pandemonium",
+        "Pandemonium - Blue skull key",
+        "House of Pain",
+        "House of Pain - Blue skull key",
+        "House of Pain - Yellow skull key",
+        "House of Pain - Red skull key",
+        "Unholy Cathedral",
+        "Unholy Cathedral - Blue skull key",
+        "Unholy Cathedral - Yellow skull key",
+        "Mt. Erebus",
+        "Mt. Erebus - Blue skull key",
+        "Limbo",
+        "Limbo - Blue skull key",
+        "Limbo - Red skull key",
+        "Limbo - Yellow skull key",
+        "Dis",
+        "Warrens",
+        "Warrens - Blue skull key",
+        "Warrens - Red skull key",
+        "Backpack",
+        "Shotgun",
+        "Rocket launcher",
+        "Plasma gun",
+        "Chaingun",
+    ]
+
+    # Specific Case for E1M4, item you have 1 shot to get. Lets not put progressive in there.
+    forbid_items(world.get_location("Command Control - Supercharge", player), progression_items)
```

Comment:
If you don't want to allow progression items in these locations then you should probably forbid any advancement items for any games? And possibly also useful items? You would do this by setting `location.progress_type = LocationProgressType.EXCLUDED`

Reply:
Oh yes, good point. I would have soft locked many games with this!